frontend/src/metabase/dashboard/components/DashCard/DashCardParameterMapper/DashCardCardParameterMapper.jsx

```diff
@@ -87,13 +83,18 @@ export function DashCardCardParameterMapper({
   dashcard,
   editingParameter,
   target,
-  mappingOptions,
   metadata,
   setParameterMapping,
   isMobile,
 }) {
   const [isDropdownVisible, setIsDropdownVisible] = useState(false);
 
+  const mappingOptions = useMemo(
```

Comment:
@WiNloSt I see in git blame that you have worked on a bug in this area. Do you have in mind any potential problem which useMemo can cause (instead of `getParameterMappingOptions` selector)?

Reply:
I've tried to follow the code, so the difference is only that this version has different `card` and `dashcard`.

While previously both are from `getCard` and `getDashCard` respectively. Now they're from the props which come from `getVisibleCards`

https://github.com/metabase/metabase/blob/4ac58f9ded42c405bc43e9ca098d98116fa43337/frontend/src/metabase/dashboard/components/DashboardGrid.tsx#L326-L342

I haven't gone into details about how different these `card` and `dashcard` are. But maybe you should add some comments here, on why this is necessary.

Reply:
nope, card is the same as before, it comes from`[dashcard.card].concat(dashcard.series || []`

Reply:
I probably don't follow how the card appears in the component, but how is the selector different to direct prop passing?

https://github.com/metabase/metabase/blob/a8e8e9e68464775373060946b4f52af69cc4106e/frontend/src/metabase/dashboard/selectors.ts#L291-L292

Reply:
Well, you're right I haven't carefully looked at the selector implementations, so both `getCard` and `getDashCard` values are still coming from the props passed to the selector. That means all parameters are the same.

I thought initially, the reason for using `useMemo` was because you want to pass different props. I'm not sure why this is different from using the selector directly (as before the change) though, because the selector `getParameterMappingOptions` is declared using `createSelector` from `@reduxjs/toolkit` which should produce a memoized selector if I'm not mistaken.

Reply:
that's the key 😄 we're working with selectors in terms of `dashboard`. so whenever mapping is changed, all connected components are re-triggered, and `state` in our case is a dashboard, which recalculates everything. Do you have a better idea how to make memoization working with selectors?

Reply:
I don't have any answers on top of my head. Optimizing rendering isn't my expert. I'd need to test this out first. But to answer your first question again. I don't think this will ever cause any problem for sure.